simplemonitor/Monitors/remote_ssh.py

```diff
@@ -0,0 +1,135 @@
+            self.get_config_option("result_type", required=True, allowed_values=[OperatorType.INTEGER.value, OperatorType.STRING.value]),
+        )
+        match self.result_type:
+            case OperatorType.INTEGER.value:
+                self.target_value = cast(int, self.get_config_option("target_value", required=True))
+            case OperatorType.STRING.value:
+                self.target_value = cast(str, self.get_config_option("target_value", required=True))
+
+    def run_test(self) -> bool:
+        # run remote command
+        client = paramiko.SSHClient()
+        client.set_missing_host_key_policy(paramiko.AutoAddPolicy)
+        try:
+            client.connect(
+                self.target_hostname,
+                username=self.ssh_username,
+                key_filename=self.ssh_private_key_path,
+                port=self.target_port,
+            )
+        except TimeoutError:
+            return self.record_fail(f"connection to {self.target_hostname} timed out")
+        except ConnectionRefusedError:
+            return self.record_fail(f"connection to {self.target_hostname} actively refused")
+        except Exception as e:
+            return self.record_fail(f"connection to {self.target_hostname} failed: {e}")
+        else:
+            _, stdout, _ = client.exec_command(self.command)
+
+        # extract and cast the actual value
+        command_result = stdout.read().decode("utf-8")  # let's hope for the best
+        client.close()  # it's only now that we are done with the client
+        actual_value = re.match(self.regex, command_result).groups()[0]
+        match self.result_type:
+            case OperatorType.INTEGER.value:
+                actual_value = cast(int, actual_value)
+            case OperatorType.STRING.value:
+                actual_value = cast(str, actual_value)
+
+        # assess the comparison logic. str and int can be checked for equality, only int can be checked for greater/less than
+        test_succeeded = False  # better be pessimistic
+        match self.operator:
+            case Operator.EQUALS.value:
+                test_succeeded = actual_value == self.target_value
+            case Operator.NOT_EQUALS.value:
+                test_succeeded = actual_value != self.target_value
+            case Operator.GREATER_THAN.value:
+                test_succeeded = actual_value > self.target_value
+            case Operator.LESS_THAN.value:
+                test_succeeded = actual_value < self.target_value
+        if self.result_type == OperatorType.STRING.value and (self.operator in [Operator.GREATER_THAN.value, Operator.LESS_THAN.value]):
+            logger.warning(f"strings compared with '{self.operator}'")
+        if test_succeeded:
+            return self.record_success(self.success_message.format(actual_value))
+        else:
+            return self.record_fail(f"actual value: {actual_value} | operator: {self.operator} | target value: {self.target_value}")
```

Comment:
Any value in letting the user also customise the failure message in a similar fashion to the success one?

Reply:
This will require some adjustments - I will have to test how doable it is to pass the format to the actual logging message (right now there is just one "placeholder" (`{}`) for `actual_value` in the configuration string)